### PR TITLE
Pv2 913 pv2 557 pv2 486 pv2 531 pv2 532 pv2 533 pv2 535 pv2 537 pv2 540 pv2 541 pv2 542 pv2 543 english and maths

### PR DESCRIPTION
**Pull request by carlbm**

### CODE REVIEW

**Comment by carlbm** [comment]
The tests have been updated to test ACT2 as well as ACT1

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentEventMatcher.cs [hunk 1]
```diff
@@ -157,17 +157,25 @@ protected override IList<ProviderPaymentEvent> GetExpectedEvents()
...
+                expected.TransactionType == actual.TransactionType &&
+                expected.AmountDue == actual.AmountDue &&
+                expected.CollectionPeriod.Period == actual.CollectionPeriod.Period &&
+                expected.CollectionPeriod.AcademicYear == actual.CollectionPeriod.AcademicYear &&
+                expected.DeliveryPeriod == actual.DeliveryPeriod &&
+                expected.Learner.ReferenceNumber == actual.Learner.ReferenceNumber &&
+                expected.Learner.Uln == actual.Learner.Uln &&
+                expected.LearningAim.StandardCode == actual.LearningAim.StandardCode)
+            {
+                if (actual.LearningAim.Reference.Equals("ZPROG001", StringComparison.OrdinalIgnoreCase))
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentEventMatcher.cs]: Would this be easier done by removing a line that populates AccountId for SfaFullyFunded events?

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentModelMatcher.cs [hunk 2]
```diff
@@ -81,17 +82,26 @@ protected override IList<PaymentModel> GetExpectedEvents()
...
+                expected.DeliveryPeriod == actual.DeliveryPeriod &&
+                expected.TransactionType == actual.TransactionType &&
+                expected.ContractType == actual.ContractType &&
+                expected.FundingSource == actual.FundingSource &&
+                expected.Amount == actual.Amount &&
+                expected.LearnerReferenceNumber == actual.LearnerReferenceNumber &&
+                expected.Ukprn == actual.Ukprn &&
+                expected.LearningAimStandardCode == actual.LearningAimStandardCode)
+            {
+                if (actual.LearningAimReference.Equals("ZPROG001", StringComparison.OrdinalIgnoreCase))
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentModelMatcher.cs]: Would this be easier done by passing null into ToPaymentModel for SfaFullyFundedPayments?

File: src/SFA.DAS.Payments.EarningEvents.Application/Mapping/FunctionalSkillEarningEventBuilder.cs [hunk 3]
```diff
@@ -23,11 +26,19 @@ public List<FunctionalSkillEarningsEvent> Build(ProcessLearnerCommand learnerSub
             foreach (var intermediateLearningAim in intermediateResults)
             {
                 var learnerWithSortedPriceEpisodes = intermediateLearningAim.CopyReplacingPriceEpisodes(intermediateLearningAim.PriceEpisodes);
-                results.Add(mapper.Map<FunctionalSkillEarningsEvent>(learnerWithSortedPriceEpisodes));
+                var functionalSkillEarning = mapper.Map<FunctionalSkillEarningsEvent>(learnerWithSortedPriceEpisodes);
+                if (learnerSubmission.Learner.PriceEpisodes.First().PriceEpisodeValues.PriceEpisodeContractType.Equals("Levy Contract", StringComparison.OrdinalIgnoreCase))
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.EarningEvents.Application/Mapping/FunctionalSkillEarningEventBuilder.cs]: Please use contsants. We already have one defined in ApprenticeshipContractTypeEarningsEventFactory

Comment (carlbm) [reply src/SFA.DAS.Payments.EarningEvents.Application/Mapping/FunctionalSkillEarningEventBuilder.cs]: Done

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentEventMatcher.cs (hunk 1 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentEventMatcher.cs]: This seems the only place where only a single change will do

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentModelMatcher.cs (hunk 2 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/EventMatchers/ProviderPaymentModelMatcher.cs]: This seems the only place where only a single change will do

**Pull request merged**